frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsTable/DynamicConversionGoalControls.tsx

```diff
@@ -0,0 +1,127 @@
+import { useActions, useValues } from 'kea'
+import { useCallback, useRef, useState } from 'react'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { IconChevronDown, IconChevronRight } from '@posthog/icons'
+
+import { ConversionGoalFilter } from '~/queries/schema/schema-general'
+import { objectsEqual } from 'lib/utils'
+
+import { marketingAnalyticsLogic } from '../../logic/marketingAnalyticsLogic'
+import { ConversionGoalDropdown } from '../common/ConversionGoalDropdown'
+import { defaultConversionGoalFilter } from '../settings/constants'
+import { LemonInput } from '@posthog/lemon-ui'
+import { marketingAnalyticsSettingsLogic } from '../../logic/marketingAnalyticsSettingsLogic'
+
+export const DynamicConversionGoalControls = (): JSX.Element => {
+    const { setDynamicConversionGoal, clearDynamicConversionGoal } = useActions(marketingAnalyticsLogic)
+    const { dynamicConversionGoal } = useValues(marketingAnalyticsLogic)
+    const { addOrUpdateConversionGoal } = useActions(marketingAnalyticsSettingsLogic)
+    const { conversion_goals } = useValues(marketingAnalyticsSettingsLogic)
+
+    const conversionGoalIdRef = useRef<string>()
+    if (!conversionGoalIdRef.current) {
+        conversionGoalIdRef.current = crypto.randomUUID()
```

Comment:
I generally prefer using UUIDv7 because you ge the extra info of when it was created, and we should have a function for that